File: bkr_services/src/main/java/org/bkr/services/service/interfaces/TemplateDetailService.java
package org.bkr.services.service.interfaces;

import java.util.List;
import java.util.Set;

import org.bkr.models.TemplateDetail;

public interface TemplateDetailService {

	public TemplateDetail save(TemplateDetail td);
	public Set<TemplateDetail> save(Set<TemplateDetail> s);
	public List<TemplateDetail> findByTemplateId(long templateId);
	public void delete(Long id);
	public TemplateDetail findById(long id);
	public long countByTemplateIdAndBreadId(long templateId,long breadId);
	public long countByBreadId(long breadId);
}

File: bkr_web/src/main/resources/public/secure/services/templatedetailservice.js
app.service("TemplateDetailService",function($http,ConfigService){
	var url=ConfigService.getBaseUrl()+"/tdetail";
	
	this.countByBreadId=function(id,successCallBack,failCallBack)
	{
		$http.get(url+"/countB",{
			params:{"y":id}
		})
		.then(function(response){successCallBack(response.data);})
		.catch(function(response){failCallBack(response);});
	}
	
	this.countByTemplateIdAndBreadId=function(data,successCallBack,failCallBack)
	{
		$http.get(url+"/countB",{
			params:
			{
				"y":data.templateId,
				"z":data.breadId
			}
		})
		.then(function(response){successCallBack(response.data);})
		.catch(function(response){failCallBack(response);});
	}
	
	this.delete=function(data,successCallBack,failCallBack){
		
		
		$http.delete(url+"/delete",{
			params:{"y":data.id}
		})
		.then(function(response){successCallBack(response.data);})
		.catch(function(response){failCallBack(response.data.message);});
			
		
		
	}
	
})